kimchi/src/circuits/domains.rs

```diff
@@ -21,24 +23,34 @@ impl<F: FftField> EvaluationDomains<F> {
     /// and `d8` (of size `8n`). If generator of `d8` is `g`, the generator
     /// of `d4` is `g^2`, the generator of `d2` is `g^4`, and the generator of `d1` is `g^8`.
     // TODO(mimoo): should we instead panic/return an error if any of these return None?
-    pub fn create(n: usize) -> Option<Self> {
-        let n = Domain::<F>::compute_size_of_domain(n)?;
+    pub fn create(n: usize) -> Result<Self, ProverError> {
+        let n = Domain::<F>::compute_size_of_domain(n)
+            .ok_or(ProverError::Prover("couldn't compute size of domain"))
+            .unwrap();
 
-        let d1 = Domain::<F>::new(n)?;
+        let d1 = Domain::<F>::new(n)
+            .ok_or(ProverError::Prover("could not construct domain"))
+            .unwrap();
```

Comment:
since the function returns a `Result`, return an error instead of unwrapping here (using `?`)

Reply:
the `ok_or` turn your Option into an Result though